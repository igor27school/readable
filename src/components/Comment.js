import React, { Component } from 'react'
import { connect } from 'react-redux'
import { COMMENT_TYPE } from '../utils/Helper'
import Voter from './Voter'

class Comment extends Component {
  render() {
    const { comments, commentId } = this.props
    const comment = comments.byId && comments.byId[commentId]
    if (!comment){
      return (
        <div>Invalid comment id: {commentId}</div>
      )
    }
    return (
      <div>
        <h4>{comment.body}
          <span> Score: {comment.voteScore}</span>
          <Voter componentType={COMMENT_TYPE} id={commentId}/>
        </h4>
      </div>
    )
  }
}

function mapStateToProps ({ comments }) {
  return {comments}
}

export default connect(mapStateToProps)(Comment)
