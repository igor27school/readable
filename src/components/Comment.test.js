import React from 'react'
import ShallowRenderer from 'react-test-renderer/shallow'
import { Comment } from './Comment'

jest.mock('../utils/ServerAPI')

describe('Comment tests', () => {
  const setup = () => {
    const props = {
      comment: {
        id: 'testId',
        body: 'test body',
        author: 'test author',
        timestamp: 1234,
        voteScore: 2,
        deleted: false,
        parentId: 'testPostId',
      },
      commentId: 'testId',
      category: 'testCategory',
    }
    const renderer = new ShallowRenderer()
    return {
      props,
      renderer
    }
  }

  it('Comment matches the snapshot', () => {
    const { renderer, props } = setup()
    renderer.render(<Comment {...props}/>)
    const result = renderer.getRenderOutput()
    expect(result).toMatchSnapshot()
  })
})
