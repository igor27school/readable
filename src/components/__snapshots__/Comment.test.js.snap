// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Comment tests Comment matches the snapshot 1`] = `
<div>
  <Link
    replace={false}
    to="/edit/testCategory/testId"
  >
    Edit
  </Link>
  <Connect(Deleter)
    id="testId"
    objectType="COMMENT_TYPE"
  />
  <p>
    Body: 
    test body
  </p>
  <p>
    Author: 
    test author
  </p>
  <p>
    Last modified: 
    Wed Dec 31 1969 16:00:01 GMT-0800 (PST)
  </p>
  <p>
    Score: 
    2
    <Connect(Voter)
      id="testId"
      objectType="COMMENT_TYPE"
    />
  </p>
</div>
`;
